### PR TITLE
action: add docs preview link when docs change.

### PR DESCRIPTION
**Pull request by yaauie**
## Release notes

[rn-skip]

## What does this PR do?

When a pull-request is opened, or when the HEAD of an existing pull-request is changed, and the changeset modifies the docs, add a comment to the PR with a link to our docs preview.

## Why is it important/What is the impact to the user?

No user impact.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files (and/or docker env variables)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by yaauie** [comment]
A couple thoughts.

In _theory_ the `synchronize` type will cause this action to add a new comment each time the PR is updated with docs changes, but it is unclear if that is (a) when the sum of the diff from `BASE...HEAD` includes changes, or (b) if the synchronize event contains _new_ changes to the docs. I included this step so that PRs created without docs changes, that later get docs changes, will also get a preview link.

We could likely reduce the noise a bit in practice by using a "find comment" step to only add the comment _once_ per PR, since the resulting link doesn't change as commits are added.

**Review by karenzone — COMMENTED** [review]
This is pretty darn clever!  Thank you. <3

File: .github/workflows/add-docs-preview-link.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+    paths: docs/**
+jobs:
+  docs-preview-link:
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
```
Comment (jsvd) [inline .github/workflows/add-docs-preview-link.yml:11]: nit: is this permission required? this github action only acts on pull-requests

File: .github/workflows/add-docs-preview-link.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      pull-requests: write
+    steps:
+    - name: Add Docs Preview link in PR Comment
+      uses: thollander/actions-comment-pull-request@v1
+      with:
+        message: ':page_with_curl: **DOCS PREVIEW** :sparkles: https://logstash_${{ github.event.number }}.docs-preview.app.elstc.co/diff'
```
Comment (jsvd) [inline .github/workflows/add-docs-preview-link.yml:17]:
```suggestion
        message: |
          :page_with_curl: **DOCS PREVIEW** :sparkles: https://logstash_${{ github.event.number }}.docs-preview.app.elstc.co/diff
          Note: if you get a "404!" please wait until the elasticsearch ci job finishes.
```

**Review by jsvd — APPROVED** [review]
LGTM, w/ two nitpicks:
1. should we let the user know that the url may not be ready yet until the ci job finishes?
2. this action shouldn't require the permission to write to issues.

File: .github/workflows/add-docs-preview-link.yml (hunk 1 above)
Comment (yaauie) [reply .github/workflows/add-docs-preview-link.yml:11]:
```suggestion
```

**Pull request merged**